### DISCUSSION TITLE
How to pass a callback from `_app.js` to children?

### DISCUSSION THREAD

**Discussion by florianwalther-private** [vercel/next.js#37880] [Help] [answered]
In my vanilla React project, I have some callbacks at the app level because multiple components in the app share it. How can I do this with my `_app.js` file?

**Comment by icyJoseph** [comment, marked as answer]
```tsx
import "../styles/globals.css";
import type { AppProps } from "next/app";
import { useState } from "react";

function MyApp({ Component, pageProps }: AppProps) {
  const [count, setCount] = useState(0);

  return (
    <div>
      <p>count is {count}</p>
      <Component {...pageProps} inc={() => setCount((x) => x + 1)} />
    </div>
  );
}

export default MyApp;
```

and then

```tsx
import type { NextPage } from "next";
import Link from "next/link";

const Home: NextPage<{ inc: () => void }> = ({ inc }) => {
  return (
    <div>
      Home <Link href="/blog">blog</Link>
      <div>
        <button onClick={inc}>+1</button>
      </div>
    </div>
  );
};

export default Home;
```

I am not sure if there's any implications though, but for example a Context Provider works fine too, but might be overkill 😅 - It's just JavaScript, as they say.

**Comment by florianwalther-private** [comment]
@icyJoseph Oh, I didn't know it was that easy! I'm actually using TypeScript but it seems to work as well. Thank you!

**Reply by icyJoseph** [reply]
Hehe my snippet was also TypeScript. 

The "It's just JavaScript" is a catch phrase used to rope people into these new technologies, like "React is just JavaScript".

Good that it worked out for you.
